### PR TITLE
Add macro functionality to tn-cli.

### PR DESCRIPTION
**Pull request by aforge**
Macros are high-level commands that expand to a sequence of
basic Tinode CLI commands.

### CODE REVIEW

**Comment by or-else** [comment]
It would also be nice to have a sample script like https://github.com/tinode/chat/blob/master/tn-cli/sample-script.txt but with macros.

**Comment by aforge** [comment]
Made `macro` import optional and introduced a sample script.

File: tn-cli/sample-macro-script.txt [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+# Create user 'Test User'
+.must $user useradd --name 'Test User' --password test123 --comment test0 \
+  --cred email:test@example.com --avatar ./test-128.jpg --tags test,test-user \
+  --auth=JRWPA --anon=JW test
+
+# Login as xena.
+.must $xena login --scheme=basic --secret=xena:xena123
+
+# Change test's publis name.
```
Comment (or-else) [inline tn-cli/sample-macro-script.txt]: `s/publis/public`

File: tn-cli/sample-macro-script.txt [hunk 2]
```diff
@@ -0,0 +1,46 @@
+# This script shows what can be done with tn-cli macros. Run it as
+#
+#   python tn-cli.py --no-login < sample-macro-script.txt
+#
+# The script performs the following:
+#
+#  - Logs in as Xena (user has root privileges)
+#  - Creates a new user Test User with useradd
```
Comment (or-else) [inline tn-cli/sample-macro-script.txt]: First creates Test then logs in as xena.

File: tn-cli/sample-macro-script.txt [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+#  - Changes Test User's name to 'Test User 1'
+#  - Modifies Test User's default anon acs to JR with chacs
+#  - Sets Test User's password to test456
+#  - Deletes Test User with userdel
+#  - Uses resolve to locate user Alice
+#  - Subscribes to Alice
+#  - Sends a message to Alice
+
+# Create user 'Test User'
```
Comment (or-else) [inline tn-cli/sample-macro-script.txt]: Maybe add an explanation that `.must` stops execution if command fails and and `$user` is a variable which will hold the result of the execution.

File: tn-cli/tn-cli.py [hunk 4]
```diff
@@ -10,6 +10,7 @@
 import grpc
 import json
 from PIL import Image
+import imp
```
Comment (or-else) [inline tn-cli/tn-cli.py]: I don't think it's currently being used. When I suggested `imp` I meant something like a command line parameter for optionally loading a macros file.
```
./tn-cli.py --load-macro=./path/to/macro.py
```

File: tn-cli/sample-macro-script.txt (hunk 3 above)
Comment (aforge) [reply tn-cli/sample-macro-script.txt]: Done.

File: tn-cli/sample-macro-script.txt (hunk 1 above)
Comment (aforge) [reply tn-cli/sample-macro-script.txt]: Done.

File: tn-cli/tn-cli.py (hunk 4 above)
Comment (aforge) [reply tn-cli/tn-cli.py]: Ah, makes sense. Thanks. Added the flag.

**Pull request merged**